### PR TITLE
EDU-1395 Extend search page

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-1395

### CODE REVIEW

File: src/components/pages/search.yml [hunk 1]
```diff
@@ -16,3 +8,8 @@ search:
   lastUpdate:
     en: Last update
     de: Letzte Änderung
+  totalItems:
+    en: |
+      {count, plural, =1 {# result} other {# results}} found
+    de: |
+      {count, plural, =1 {# Ergebnis} other {# Ergebnisse}} gefunden
```
Comment (ahelmberger) [inline src/components/pages/search.yml:15]: I know it was like that before, but the pipe syntax should actually only be used for multiline texts (as it creates a line break after each line). It should be written like this:

~~~yml
  totalItems:
    en: "{count, plural, =1 {# result} other {# results}} found"
    de: "{count, plural, =1 {# Ergebnis} other {# Ergebnisse}} gefunden"
~~~

File: src/services/media-library-service.js [hunk 2]
```diff
@@ -116,6 +108,29 @@ class MediaLibraryService {
...
+      textAndResourceTypeQueryConditions.push({ resourceType: { $in: resourceTypes } });
+    }
+
+    const mediaLibraryItems = await this.mediaLibraryItemStore.getMediaLibraryItemsByConditions(textAndResourceTypeQueryConditions);
+
+    const positiveTokensArray = [...textQuery.positiveTokens].filter(token => token.toLowerCase());
+    return mediaLibraryItems
+      .map(item => {
+        const exactTagMatchCount = item.tags.filter(tag => textQuery.positiveTokens.has(tag.toLowerCase())).length;
+        const partialNameMatchCount = positiveTokensArray.filter(token => item.name.toLowerCase().includes(token)).length;
```
Comment (ahelmberger) [inline src/services/media-library-service.js:128]: Now we also calculate a relevance based on the file name even for the case when we only search by tags, as this function is used in both `getSearchableMediaLibraryItemsByTags` and `getSearchableMediaLibraryItemsByTagsOrName`. Maybe we could refactor this function to be like `_getSearchableMediaLibraryItems({ textQuery, resourceTypes, searchAlsoInNames = false })` and, depending on that, the function does the query and relevance calculation?

File: src/components/pages/search.yml (hunk 1 above)
Comment (adrianaluchian) [reply src/components/pages/search.yml]: Indeed, thanks for pointing this out. I've corrected this and all such misuse cases.

File: src/services/media-library-service.js (hunk 2 above)
Comment (adrianaluchian) [reply src/services/media-library-service.js]: Makes sense, corrected in this way, thanks!

Comment (ahelmberger) [reply src/services/media-library-service.js]: Nice, looks good!

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**